src/ai/backend/storage/api/vfolder/types.py

```diff
@@ -0,0 +1,274 @@
+from ai.backend.storage.types import CapacityUsage, TreeUsage
+
+__all__ = (
+    "VolumeIDModel",
+    "VolumeInfoModel",
+    "VolumeInfoListModel",
+    "VFolderIDModel",
+    "VFolderInfoRequestModel",
+    "VFolderInfoModel",
+    "VFolderCloneModel",
+    "QuotaIDModel",
+    "QuotaScopeInfoModel",
+    "QuotaConfigModel",
+)
+
+
+class BaseModel(PydanticBaseModel):
+    """Base model for all models in this module"""
+
+    model_config = {"arbitrary_types_allowed": True}
+
+
+VolumeID: TypeAlias = uuid.UUID
+
+
+# Common fields for VolumeID and VFolderID
+VOLUME_ID_FIELD = Field(
+    ...,
+    validation_alias=AliasChoices(
+        "vid", "volumeid", "volume_id", "VolumeID", "Volume_Id", "Volumeid"
+    ),
+)
+VFOLDER_ID_FIELD = Field(
+    ...,
+    validation_alias=AliasChoices(
+        "vfid", "vfolderid", "vfolder_id", "VFolderID", "VFolder_Id", "VFolderid"
+    ),
+)
+QUOTA_SCOPE_ID_FIELD = Field(
+    ...,
+    validation_alias=AliasChoices(
+        "qsid",
+        "quotascopeid",
+        "quota_scope_id",
+        "QuotaScopeID",
+        "Quota_Scope_Id",
+        "QuotaScopeid",
+        "Quota_ScopeID",
+        "Quota_Scopeid",
+        "quotaScopeID",
+        "quotaScopeid",
+    ),
+)
+
+
+class VolumeIDModel(BaseModel):
+    volume_id: VolumeID = VOLUME_ID_FIELD
+
+    @model_validator(mode="before")
+    def validate_all_fields(cls, values: dict[str, Any]) -> dict[str, Any]:
+        if "volume_id" in values and not isinstance(values["volume_id"], uuid.UUID):
+            raise ValueError("volume_id must be a UUID")
+        return values
+
+
+class VolumeInfoModel(BaseModel):
+    """For `get_volume`, `get_volumes` requests"""
+
+    volume_id: VolumeID = VOLUME_ID_FIELD
+    backend: str = Field(...)
+    path: Path = Field(...)
+    fsprefix: PurePath | None
+    capabilities: list[str] = Field(...)
+    options: Mapping[str, Any] | None
+
+    @model_validator(mode="before")
+    def validate_all_fields(cls, values: dict[str, Any]) -> dict[str, Any]:
+        if "volume_id" in values and not isinstance(values["volume_id"], uuid.UUID):
+            raise ValueError("volume_id must be a UUID")
+        if "backend" in values and not isinstance(values["backend"], str):
+            raise ValueError("backend must be a string")
+        if "path" in values and not isinstance(values["path"], Path):
+            raise ValueError("path must be a Path object")
+        if values.get("fsprefix") is not None and not isinstance(values["fsprefix"], PurePath):
+            raise ValueError("fsprefix must be a PurePath or None")
+        if "capabilities" in values and not isinstance(values["capabilities"], list):
+            raise ValueError("capabilities must be a list of strings")
+        if values.get("options") is not None and not isinstance(values["options"], Mapping):
+            raise ValueError("options must be a mapping or None")
+        return values
```

Comment:
Pydantic already provides built-in validation support. What was the reason for defining it separately?